design-book/src/design-paradigms.md

```diff
@@ -0,0 +1,28 @@
+# Design Paradigms
+
+## Non-Overlapping
+
+The Ui should enable you to always view all relevant information and tools at any given moment, you should never have to move a window out of your way, for that reason, as well as sticking with bevy's modular structure, Bevy uses a divided window layout we call Panes.
+
+1. Screens; the entire window enables you to configure workspaces using multiple editors.
+2. Panes; the container of an editor. Editors can each operate similar to a stand alone editor, like for modeling, painting or scripting. Editors can both follow Screen level user context ("Active" or "Selected") and offer local context browsing.
+3. Regions; every editor allows further subdivision to provide button/menu headers, toolbars, channel views, and what more is needed to serve its functionality.
+
+## Non-Blocking
+
+The UI should stay responsive whenever possible. Tools and interface options should not block the user from using any other parts of Bevy Editor. When long-running blocking tasks are required, it should be clearly indicated and allow an immediate exit. Users don't like their workflow being interrupted!
+
+## Non-Modal
+
+User input should remain as consistent and predictable as possible, do not change what inputs do on the fly without clear communication.
+
+When input behavior changes (e.g. a special camera mode that overrides normal controls), the change in behavior should be temporary, and require continuous user input (typically pressing a key or holding a click) to remain active.
+
+## Select -> Operate
+
+In [Blender](https://developer.blender.org/docs/features/interface/human_interface_guidelines/paradigms/#select-operate) you first indicate which data you work on, and then what you want to do. The same applies for Bevy. Quote from Blender's guidelines: "This follows the non-modal principle; there's no active tool mode you need to set first to be able to use the tool on what you select after. This concept enables a fast and flexible work flow."
```

Comment:
My very naive non UX person take :D

Perhaps there is some jargon that I am not familiar with, because I instantly thought of Blender when I read the "Non-Modal" subheading.

Is this more of an operation/interaction specific "logical order"?

Tools like "terrain editors" would also have some kind of "edit" mode. Would this design paradigm include those?

Could ask the same about will Bevy editor not have a "Play" mode?

For context, Blender itself even uses the word "mode" prominently.

<img width="161" alt="image" src="https://github.com/user-attachments/assets/90ce7680-ee25-498f-999d-df649a16a2b1">

and

<img width="429" alt="image" src="https://github.com/user-attachments/assets/59615de2-4b23-4e1d-9e67-67b1a6366dcf">

Reply:
Blenders non modal does have exceptions.
I left it open ended so that when we discover a necessary exception we can add it.

https://developer.blender.org/docs/features/interface/human_interface_guidelines/paradigms/#non-modal